apps/web/screens/Project/Project.tsx

```diff
@@ -108,20 +109,42 @@ const ProjectPage: Screen<PageProps> = ({ projectPage, news, namespace }) => {
   })
 
   const navigationList: NavigationItem[] = projectPage.sidebarLinks.map(
-    ({ primaryLink, childrenLinks }) => ({
-      title: primaryLink.text,
-      href: primaryLink.url,
-      active: router.asPath === primaryLink.url,
-      items: childrenLinks.map(({ text, url }) => ({
-        title: text,
-        href: url,
-      })),
-    }),
+    ({ primaryLink, childrenLinks }) => {
+      let isAnyChildActive = false
+      const items = childrenLinks.map(({ text, url }) => {
```

Comment:
a `.map()` function with side effects is generally considered an anti-pattern. i wonder if you'd actually just want to create either some helpers to move these "is active" logics to some shared place (even if in the same file)

and it might also be helpful to actually create a more useful type to begin with for all these items of things with a title and an active state.
```
interface NavItemWithActiveState extends NavigationItem {
  isActive: boolean
}
```
and then you could just map all the items to this type in a single pass.

this means that eg. the logic here would seem much simpler to read at least

Reply:
These map calls were in fact both mapping the items and calculating the active flag at the same time, but not necessarily easy to read through. This logic has now been extracted to helpers to improve readability of the code.  The `NavigationItem` model already contains an `active` field, so there's no need to create such interface as you suggested.